### DISCUSSION TITLE
Conditional Logic Result not being read as number

### DISCUSSION THREAD

**Discussion by glitchy-sage** [grafana/grafana-json-datasource#197] [Q&A]
Hi,

Hoping you can assist, I've been using this plugin to pull data from our ThousandEyes API and graph it.

While playing around with the best ways to make interactive dashboards, I've come across a small issue with the JSONata interpretation by Grafana.

Explained:
Below is a screenshot of my query
![image](https://user-images.githubusercontent.com/79307428/140277977-bb1fdfbc-f1d9-4438-823a-984a8cdabed1.png)

at:
1 & 3 - you will see me using conditional logic, this is because depending on the type of Test we select, we may see a metric present for 1 or not present for the other. 

In simple terms, a metric like sslTime, is sometimes present in the json data, and sometimes it is not. To get around this I used conditional logic. (On our Table this is not seen as a number, and cannot have units applied to it or be graphed on separate panels)

2 - you will see me using a plain query instead of conditional logic. (On our Table this is seen as a number and is able to have a unit applied to it)

Below is a screenshot of the table, you will see DNS Time has (ms) applied to it, this is because only this query result is being seen as a number, being able to be graphed.
![image](https://user-images.githubusercontent.com/79307428/140278470-8547325c-765d-4de1-b450-e848721508c0.png)

I am unsure if this is an expected result using conditonal logic, but if there is a way to fix this or work around it, 
I would be grateful for assistance.

Regards,
Me

**Comment by marcusolsson** [comment]
I'm not certain that I fully understand your question. 

- The DNS Time is parsed as a number, because the Type is set to `Number`. 
- The ms comes from your Unit configuration.

Are you saying that missing values are interpreted as `0`?

**Reply by glitchy-sage** [reply]
Apologies, will attempt to explain a bit better.

So I've set all of my Types to Number and have set all of my units to be 'ms'.

From what I can see, when I query without using any conditional logic, it works perfectly.
E.g. web.httpServer[agentName = $AgentNames[0]].dnsTime 
which returns a metric that is able to be graphed and have unit standards applied (ms)

but when I attempt it with a conditional logic expression, the data still pulls, but it cannot be graphed or have unit standards applied.
It almost seems like Grafana is not seeing it as a number.
e.g. web.httpServer[agentName = $AgentNames[0]].[$exists(dnsTime) ? dnsTime: 0]
which returns a metric which cannot be graphed or have unit standards applied.

Just to give an example of more or less the same call:
![image](https://user-images.githubusercontent.com/79307428/140333472-8cd64bb2-4603-486b-89cf-3e8919a55ab0.png)


On the graph this shows as:
![image](https://user-images.githubusercontent.com/79307428/140333548-6e8ed95e-8963-4fa5-9563-076a4b802bb5.png)

I am unsure what causes this, 
I have also tried to add the $number() function in the conditional expression, but this does not work.
E.g. web.httpServer[agentName = $AgentNames[0]].[$exists(dnsTime) ? $number(dnsTime) : 0]

So from what I can see, values do not come in as 0, but are not seen as numbers (Only when applying conditional formatting)

Note: I am still very new to jsonata, if there is a way to check for a key's existence a different way, this would assist.

**Reply by marcusolsson** [reply]
Could you maybe share an example of an (anonymized) JSON response so that I can replicate this myself?

**Reply by glitchy-sage** [reply]
Hi, 
100%, have an extract of the exact query I was working on.


[eg.txt](https://github.com/marcusolsson/grafana-json-datasource/files/7481228/eg.txt)
